examples/polygon-p2p/src/main.rs

```diff
@@ -0,0 +1,82 @@
+//! Example for how hook into the polygon p2p network
+//!
+//! Run with
+//!
+//! ```not_rust
+//! cargo run -p polygon-p2p
+//! ```
+//!
+//! This launch the regular reth node overriding the engine api payload builder with our custom.
+//!
+//! Credits to: <https://blog.merkle.io/blog/fastest-transaction-network-eth-polygon-bsc>
+use chain_cfg::{boot_nodes, chain_spec, head};
+use reth_network::{
+    config::NetworkMode, NetworkConfig, NetworkEvent, NetworkEvents, NetworkManager,
+};
+
+use reth_discv4::Discv4ConfigBuilder;
+use reth_provider::test_utils::NoopProvider;
+use reth_tracing::{
+    tracing::info, tracing_subscriber::filter::LevelFilter, LayerInfo, LogFormat, RethTracer,
+    Tracer,
+};
+use secp256k1::{rand, SecretKey};
+use std::{
+    net::{Ipv4Addr, SocketAddr},
+    time::Duration,
+};
+use tokio_stream::StreamExt;
+pub mod chain_cfg;
+
+#[tokio::main]
+async fn main() {
+    // The ECDSA private key used to create our enode identifier.
+    let secret_key = SecretKey::new(&mut rand::thread_rng());
+
+    let _ = RethTracer::new()
+        .with_stdout(LayerInfo::new(
+            LogFormat::Terminal,
+            "info".to_string(),
+            LevelFilter::INFO.into(),
+            Some("always".to_string()),
+        ))
+        .init();
+
+    // The local address we want to bind to
+    let local_addr = SocketAddr::new(Ipv4Addr::UNSPECIFIED.into(), 30303);
+
+    // The network configuration
+    let net_cfg = NetworkConfig::builder(secret_key)
+        .chain_spec(chain_spec())
+        .set_head(head())
+        .network_mode(NetworkMode::Work)
+        .listener_addr(local_addr)
+        .build(NoopProvider::default());
```

Comment:
I think we want a helper function for this in a followup `build_with_noop_provider`

https://github.com/paradigmxyz/reth/blob/30aa3e6792701a8719a49995d79a0525bfc33881/crates/net/network/src/config.rs#L421-L421

the provider in p2p isn't that useful to begin with